contrib/ruby/ext/trilogy-ruby/cext.c

```diff
@@ -937,6 +937,42 @@ static VALUE rb_trilogy_query(VALUE self, VALUE query)
     return execute_read_query_response(ctx);
 }
 
+static VALUE rb_trilogy_pipelined_query(VALUE self, VALUE queries)
+{
+    struct trilogy_ctx *ctx = get_open_ctx(self);
+
+    Check_Type(queries, T_ARRAY);
+
+    rb_encoding * encoding = rb_to_encoding(ctx->encoding);
+
+    long count = RARRAY_LEN(queries);
+    long index;
+    for (index = 0; index < count; index++) {
+        VALUE query = RARRAY_AREF(queries, index);
+        StringValue(query);
+
+        query = rb_str_export_to_enc(query, encoding);
+
+        int rc = trilogy_query_send(&ctx->conn, RSTRING_PTR(query), RSTRING_LEN(query));
+        if (rc == TRILOGY_AGAIN) {
+            rc = flush_writes(ctx);
+        }
+
+        if (rc < 0) {
+            handle_trilogy_error(ctx, rc, "trilogy_query_send");
+        }
+    }
+
+    VALUE results = rb_ary_new2(count);
+    for (index = 0; index < count; index++) {
+        ctx->conn.packet_parser.sequence_number = 1; // HACK
```

Comment:
I'm not sure if the sequence number can really be assumed to be always `1`. From what I could see it's the case, but we could also record the sequence number of each query.

Reply:
Won't be 1 if the query is long enough to span multiple MySQL packets.

Reply:
Right, I think we need to store these values in an array somewhere to restore them.